### PR TITLE
SDK-212 conversion icp

### PR DESCRIPTION
**Pull request by wackyleo459**
Currently pulls in latest xdr_permyriad_per_icp rate, to convert cycles into ICP amount in the CycleSlider subcomponent. 

With this change, users can infer an accurate cost when allocating cycles to canisters.

https://dfinity.atlassian.net/browse/SDK-212?atlOrigin=eyJpIjoiNTA3NzhjYzYwZTA2NDkyZmJkNzI0ZTMyZTdmMjAwMzUiLCJwIjoiaiJ9

### CODE REVIEW

File: dfx.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
       "wasm": "target/wasm32-unknown-unknown/release/wallet-opt.wasm"
     }
   },
+  "dfx": "0.8.4",
```
Comment (krpeacock) [inline dfx.json:27]: I'd rather avoid this so we don't need to keep updating the version repeatedly
```suggestion
```

File: wallet_ui/components/CycleSlider.tsx [hunk 2]
```diff
@@ -143,10 +144,22 @@ interface Props {
 
 function CycleSlider(props: Props) {
   const { balance = 0, cycles, setCycles, loading } = props;
+  const [xdr_permyriad_per_icp, setRate] = React.useState(BigInt(0));
+
+  const cyclesToIcp = React.useMemo(() => {
+    if (cycles === 0) {
+      return 0;
+    }
+    if (!xdr_permyriad_per_icp) { return null}
```
Comment (krpeacock) [inline wallet_ui/components/CycleSlider.tsx:153]: Good job on this tricky variable name!

**Review by krpeacock — CHANGES_REQUESTED** [review]
Before we merge this, can you add some documentation on where to find the XDR declarations, for the next person along who may need to upgrade this?

File: dfx.json (hunk 1 above)
Comment (wackyleo459) [reply dfx.json:27]: @krpeacock for documentation about XDR declaration, is it best to have that in the README or CHANGELOG file? or both?

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Currently pulls in live converstion rate from XDR to ICP, via a canister query method `xdr_permyriad_per_icp` from: `./wallet_ui/declarations/xdr`.
```
Comment (krpeacock) [inline CHANGELOG.md:12]: Sorry, I wasn't clear. We got the candid declaration from another location. If we need to upgrade the interface in the future, we should leave some sort of reference to how we got it so we can go fetch the latest version

**Review by krpeacock — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (wackyleo459) [reply CHANGELOG.md:12]: - Currently pulls XDR to ICP conversion rate from `nns-dapp/frontend/ts/src/canisters/cyclesMinting/canister.did` that in turn queries from CMC (cycles minting canister): "rkp4c-7iaaa-aaaaa-aaaca-cai".

Comment (wackyleo459) [reply CHANGELOG.md:12]: Currently pulls live XDR to ICP conversion rate via declaration sourced from `nns-dapp/frontend/ts/src/canisters/cyclesMinting/canister.did`. This in turn queries the cycles minting canister ("rkp4c-7iaaa-aaaaa-aaaca-cai").

**Review by krpeacock — APPROVED** [review]

**Pull request merged**